### PR TITLE
Spring security

### PR DESCRIPTION
**Pull request by vlykhvar**
> Configure Spring Security with in memory authentication.

> Replace all ?userId @RequestParam in your code with Authentication object.

> Add @NonNull, @Min, etc annotations where applicable (add this dependency to your project, see example here)

> Implement custom email validation.

> In the UserRegistrationDto add two fields: password and repeatPassword.

> Implement validation to check that password and repeatPassword are the same.

> Implement CustomGlobalExceptionHandler.

> Remove method login() in AuthenticationService, HashUtil and salt from User entity. Start using passwordEncoder in UserService -> add() for password hashing.

### CODE REVIEW

**Review by AndreiYazadzhi — CHANGES_REQUESTED** [review]
image

**Review by ivlytvynets — CHANGES_REQUESTED** [review]
Email validator isn't present here

**Review by AndreiYazadzhi — APPROVED** [review]

File: src/main/java/com/dev/theater/security/validators/EmailValidate.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import jakarta.validation.ConstraintValidatorContext;
+
+public class EmailValidate implements ConstraintValidator<ValidateEmail, String> {
+    @Override
+    public void initialize(ValidateEmail constraintAnnotation) {
+
+    }
+
+    @Override
+    public boolean isValid(String s, ConstraintValidatorContext constraintValidatorContext) {
```
Comment (Sofasmile) [inline src/main/java/com/dev/theater/security/validators/EmailValidate.java]: s - не дуже інформативна назва змінної

File: src/main/java/com/dev/theater/security/validators/EmailValidate.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        return s != null && s.matches("[@]+[.]")
```
Comment (Sofasmile) [inline src/main/java/com/dev/theater/security/validators/EmailValidate.java]: regex краще винести в константу

File: src/main/java/com/dev/theater/security/validators/PasswordValidator.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                           ConstraintValidatorContext constraintValidatorContext) {
+        Object fieldValue = new BeanWrapperImpl(userRegistrationDto)
+                .getPropertyValue(password);
+        Object fieldMatchValue = new BeanWrapperImpl(userRegistrationDto)
+                .getPropertyValue(repeatPassword);
+        if (fieldValue != null) {
+            return fieldValue.equals(fieldMatchValue);
+        } else {
+            return fieldMatchValue == null;
+        }
```
Comment (Sofasmile) [inline src/main/java/com/dev/theater/security/validators/PasswordValidator.java]: спробуй спростити дану конструкцію

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Pull request merged**